packages/jest-mock-apollo/src/GraphQLClientFactory.ts

```diff
@@ -0,0 +1,95 @@
+import {buildSchema, GraphQLSchema} from 'graphql';
+import {GraphQLRequest, ApolloLink} from 'apollo-link';
+import {
+  ApolloReducerConfig,
+  InMemoryCache,
+  IntrospectionFragmentMatcher,
+} from 'apollo-cache-inmemory';
+import ApolloClient from 'apollo-client';
+import {GraphQLMock, GraphQLClientConfig, GraphQLClientOptions} from './types';
+import MockApolloLink from './MockApolloLink';
+import Requests from './Requests';
+
+export type MockGraphQLClient = ApolloClient<any> & {
+  graphQLRequests: Requests;
+  graphQLResults: Promise<any>[];
+};
+
+function defaultGraphQLMock({operationName}: GraphQLRequest) {
+  return new Error(
+    `Can’t perform GraphQL operation '${operationName ||
+      ''}' because no mocks were set.`,
+  );
+}
+
+export default class GraphQLClientFactory {
+  private unionOrIntersectionTypes: any[];
+  private schema: GraphQLSchema;
+  private cacheOptions: ApolloReducerConfig;
+
+  constructor({
+    unionOrIntersectionTypes = [],
+    schema,
+    schemaSrc,
+    cacheOptions = {},
+  }: GraphQLClientConfig) {
+    this.unionOrIntersectionTypes = unionOrIntersectionTypes;
+
+    const proposedSchema = schemaSrc == null ? schema : buildSchema(schemaSrc);
+    if (proposedSchema == null) {
+      throw new Error('schema or schemaSrc is required.');
+    }
+    this.schema = proposedSchema;
+
+    this.cacheOptions = cacheOptions;
+
+    this.create = this.create.bind(this);
+  }
+
+  public create(
```

Comment:
Yeah, seems like we never actually mutate anything on this class or otherwise use it's instances, don't see why we need a class